gluster-exporter/metric_peer_counts.go

```diff
@@ -0,0 +1,181 @@
+	return outB, err
+}
+
+// PeerMetrics : exposes PV, LV, VG counts
+type PeerMetrics struct {
+	PVCount int
+	LVMap   map[string]int
+	VGCount int
+}
+
+type myVGDetails struct {
+	LVUUID  string `json:"lv_uuid"`
+	LVName  string `json:"lv_name"`
+	PoolLV  string `json:"pool_lv"`
+	VGName  string `json:"vg_name"`
+	LVPath  string `json:"lv_path"`
+	LVCount string `json:"lv_count"`
+	PVCount string `json:"pv_count"`
+}
+
+// NewPeerMetrics : provides a way to get the consolidated metrics (such PV, LV, VG counts)
+func NewPeerMetrics() (*PeerMetrics, error) {
+	cmdStr := "lvm vgs --noheading --reportformat=json -o lv_uuid,lv_name,pool_lv,vg_name,lv_path,lv_count,pv_count"
+	outBs, err := execCommand(cmdStr)
+	if err != nil {
+		return nil, err
+	}
+	dec := json.NewDecoder(bytes.NewReader(outBs))
+	var vgs []myVGDetails
+	var vgMap = make(map[string]myVGDetails)
+	for {
+		t, decErr := dec.Token()
+		if decErr == io.EOF {
+			break
+		}
+		if decErr != nil {
+			err = errors.New("Unable to parse JSON output: " + err.Error())
+			return nil, err
+		}
+		// if the token is 'vg', collect/decode the details into VGDetails array
+		if t == "vg" {
+			decErr = dec.Decode(&vgs)
+		}
+		if decErr != nil {
+			err = errors.New("JSON output changed, parse failed: " + err.Error())
+			return nil, err
+		}
+	}
+	for _, vg := range vgs {
+		// collect the unique vgs into the map
+		if _, ok := vgMap[vg.VGName]; !ok {
+			vgMap[vg.VGName] = vg
+		}
+	}
+	pMetrics := &PeerMetrics{VGCount: len(vgMap), LVMap: make(map[string]int)}
+	for k, vg := range vgMap {
+		log.Printf("%s: %+v\n", k, vg)
+		if count, convErr := strconv.Atoi(strings.TrimSpace(vg.LVCount)); convErr == nil {
+			pMetrics.LVMap[vg.VGName] = count
+		}
+		if count, convErr := strconv.Atoi(strings.TrimSpace(vg.PVCount)); convErr == nil {
+			pMetrics.PVCount += count
+		}
+	}
+	return pMetrics, nil
+}
+
+var (
+	gnrlMetricLabels = []MetricLabel{
+		{
+			Name: "name",
+			Help: "Name of the metric, whose count is collected (PV or LV or VG etc)",
+		},
+		{
+			Name: "peerID",
+			Help: "Peer ID of the host on which the metrics are collected",
+		},
+	}
+	// an additional information of 'vgName' is added
+	// this specifies which Volume Group the LV count belongs
+	lvMetricLabels = []MetricLabel{
+		{
+			Name: "name",
+			Help: "Logical Volume Metric details",
+		},
+		{
+			Name: "peerID",
+			Help: "Peer ID of the host on which the LV details are picked",
+		},
+		{
+			Name: "vgName",
+			Help: "Volume Group Name, shows no: of lvs under the specified Volume Group",
+		},
+	}
+
+	glusterPVCountMetric = newPrometheusGaugeVec(Metric{
+		Namespace: "gluster",
+		Name:      "pv_count",
+		Help:      "No: of physical volumes, got through pvs command",
+		LongHelp:  "",
+		Labels:    gnrlMetricLabels,
+	})
+	glusterLVCountMetric = newPrometheusGaugeVec(Metric{
+		Namespace: "gluster",
+		Name:      "lv_count",
+		Help:      "No: of logical volumes, got through lvs command",
```

Comment:
`No of logical volumes` is enough

Reply:
Acknowledged

Reply:
this is not addressed. "Number of logical volumes"